### PR TITLE
Addition of the store API and general frontend

### PR DESCRIPTION
**Pull request by flupe**
The Store API allows to get, update and delete resources without having to bother with handwritten HTTP requests. Entities are cached in the local store to prevent repeating requests. Everything is handled with Promises so it's easy to catch errors and respond accordingly.

The following things have been added:
- User signup, login, and logout.
- Got rid of the dashboard and added pages for user groups and workspaces.
- Added user group index, group creation and group deletion.
- Added generic styles for placeholder content while resources are loading.
Most of the work on groups is temporary since a big refactor is coming up and ACL is being developped.

Things that are not yet done, but not urgent:
- When someone requests a resource in the store, we give them the plain data, and not a proxy. Ideally, we would like to provide an immutable proxy to the data, to prevent people from screwing with the store, but I did not find how to do this properly yet.
- Assuming you fetch a resource from the store while still having another version on the local store, we need to find a way to update the old version to mirror the new one (i.e in place modification, for Vue to catch up on the changes). I did not look at it much, it works good enough for now but should be looked after.
Especially when we will add message queues to notifiy of resource changes.

Things that are left unimplemented:
- I did not write a store helper for _updating_ relationships (i.e replacing the entire data array for a `to-many` relationship, or changing the content for a `to-one` relationship).
  It should be easy to implement, I just didn't have time.
- We might want to generalize relationship helpers further than the current state.
  - `rels.add` could also support `to-one` relationships (and work just as `rels.update` would for them).
  - Same remark for `rels.delete`.
  - `rels.delete` and `rels.add` take linkages as parameters, but since a linkage is only an object containing an id, perhaps change the functions to take directly ids as parameters.

## Things that need to be done before the merge:
Because I needed to add login & signup pages, that do not belong inside the default tozti layout, I changed a bit how routing works.
Meaning you cannot simply do `tozti.routes.unshift(...)` to add new routes (it will work, but your new view will not be shown inside the default layout).
Someone should add a method `tozti.addRoute` to allow this again, and prevent the same issue from happening (ie abstracting over how routes are stored).

### CODE REVIEW

File: client/components/UserPreview.vue [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <router-link :to="{ name: 'group', params: { id } }">
+    <div v-if="resource" class="group-item">
```
Comment (Lapin0t) [inline client/components/UserPreview.vue]: C'est normal que ça ne soit pas `:v-if="..."`?

Comment (flupe) [reply client/components/UserPreview.vue]: Oui.
https://vuejs.org/v2/guide/conditional.html

File: client/bootstrap.js [hunk 2]
```diff
@@ -86,6 +87,41 @@ export let tozti = window.tozti = {
...
+    let new_route = { path: route, component: component }
+    if (name != null) {
+      new_route.name = name
+    }
+    let found = false
+    // either a route with the specified meta tag exists
+    // in which case we add our widget as a children of it
+    for (let route of tozti.routes) {
+      if (!('readonly' in route)) {
+        if (route.meta === meta) {
```
Comment (flupe) [inline client/bootstrap.js]: Alors en fait `===` ça teste l'égalité sctricte entre 2 objets, mais du coup c'est une égalité physique.
A priori là ça ne retournera jamais `true`.

File: client/main.js [hunk 3]
```diff
@@ -78,50 +79,35 @@ export let tozti = window.tozti = {
...
   },
 
+
   /**
    * @param {string} name  - The name of the workspace menu item.
    * @param {string} route - The name of the route it is associated with.
    *                         This route should expect an id as a parameter.
    */
-  addWorkspaceMenuItem(name, route, props = {}) {
+  addWorkspaceMenuItem(ame, route, props = {}) {
```
Comment (poechsel) [inline client/main.js]: `name` au lieu de `ame` ?

Comment (flupe) [reply client/main.js]: good catch!

**Pull request merged**